manager/src/test/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncSegmentationAPITest.java

```diff
@@ -64,17 +70,24 @@ public void testShardSegmentation() {
         // the bridge gets assigned to the default shard
         TestUtils.createBridge(new BridgeRequest(TestConstants.DEFAULT_BRIDGE_NAME));
 
-        // The default shard retrieves the bridge to deploy
-        List<BridgeDTO> bridgesToDeployForDefaultShard = TestUtils.getBridgesToDeployOrDelete().as(new TypeRef<List<BridgeDTO>>() {
+        final List<BridgeDTO> bridgesToDeployForDefaultShard = new ArrayList<>();
+        await().atMost(5, SECONDS).untilAsserted(() -> {
```

Comment:
Since we do not trigger `Work` immediately I had to change a few tests to wait.

Reply:
Indeed. I have found a number of the tests creating `Bridge`'s\'Processor`'s to be flakey.

This `ShardBridgesSyncSegmentationAPITest` test is now annotated with `@TestProfile(WorkerSchedulerProfile.class)` that schedules `WorkManager` to execute the `Work` to provision/de-provision the `Bridge` and so we have to wait until it's done its stuff. Previously we emitted the VertX EventBus event immediately and the `Worker` would have stood greater chance of completing "in time". The test is now more robust (IMO).